client/db_test.go

```diff
@@ -0,0 +1,75 @@
+package client_test
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/server"
+)
+
+func ExampleDB() {
+	s := server.StartTestServer(nil)
+	defer s.Stop()
+
+	db := client.Open("https://root@" + s.ServingAddr())
+
+	if err := db.Put("key", "value"); err != nil {
+		log.Fatal(err)
+	}
+	result := db.Get("key")
+	fmt.Println(1, result.Rows[0].ValueStringer())
+
+	if r := db.Inc("inc", 100); r.Err != nil {
+		log.Fatal(r.Err)
+	}
+	result = db.Get("key", "inc")
+	fmt.Println(2, result.Rows[0].ValueStringer())
+	fmt.Println(3, result.Rows[1].ValueInt())
+
+	b := &client.Batch{}
+	b.Get("key")
+	b.Scan("i", "j", 100)
+	b.Put("foo", "bar")
+	if err := db.Run(b); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(4, b.Results[0].Rows[0].ValueStringer())
+	fmt.Println(5, b.Results[1].Rows[0].ValueInt())
+	fmt.Println(6, b.Results[2].Rows[0].ValueStringer())
+
+	err := db.Tx(func(tx *client.Tx) error {
+		b := &client.Batch{}
+		b.Put("aa", "1")
+		b.Put("ab", "2")
+		return tx.Commit(b)
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	result = db.Scan("a", "b", 100)
+	fmt.Println(7, result.Rows[0].ValueStringer())
+	fmt.Println(8, result.Rows[1].ValueStringer())
+
+	if err := db.Del("key", "inc"); err != nil {
+		log.Fatal(err)
+	}
+	result = db.Get("key", "aa", "inc")
+	fmt.Println(9, result.Rows[0].Exists())
```

Comment:
this would be handled nicely with my above api suggestion. If it exists, most of the time you also want the value.